Makefile

```diff
@@ -6,7 +6,7 @@ else
 endif
 
 .PHONY: build
-build: $(OUTPUT) ## Build the executable for current architecture (local dev)
+build: test $(OUTPUT) ## Build the executable for current architecture (local dev)
```

Comment:
build should not depend on test
A new target should depend on build AND test IMO